src/js/controller/DrawingController.js

```diff
@@ -247,13 +247,37 @@
   };
 
   ns.DrawingController.prototype.increaseZoom_ = function (zoomMultiplier) {
+    var coords = this.getSpriteCoordinates(this._clientX,this._clientY);
+    var off = this.getOffset();
+    var oldWidth = this.getContainerWidth_() / this.renderer.getZoom();
+    var oldHeight = this.getContainerHeight_() / this.renderer.getZoom();
+    var xRatio = (coords.x - off.x) / oldWidth;
+    var yRatio = (coords.y - off.y) / oldHeight;
     var step = (zoomMultiplier || 1) * this.getZoomStep_();
     this.setZoom_(this.renderer.getZoom() + step);
+    var newWidth = this.getContainerWidth_() / this.renderer.getZoom();
+    var newHeight = this.getContainerHeight_() / this.renderer.getZoom();
+    this.setOffset(
+      off.x + ((oldWidth - newWidth) * xRatio),
+      off.y + ((oldHeight - newHeight) * yRatio)
+    );
   };
 
   ns.DrawingController.prototype.decreaseZoom_ = function (zoomMultiplier) {
```

Comment:
Let's have a single method and remove duplication here: 

    ns.DrawingController.prototype.updateZoom_ = function (delta) {
        // your new before-code
        this.setZoom_(this.renderer.getZoom() + delta);
        // your new after-code
    }

And either use it in increase/decreaseZoom, or directly remove the increase/decreaseZoom methods and use updateZoom directly.